janus_server/src/datastore.rs

```diff
@@ -821,6 +822,74 @@ impl Transaction<'_> {
                 .await?,
         )
     }
+
+    /// If a collect job corresponding to the provided values exists, its UUID
+    /// is returned, which may then be used to construct a collect job URI. If
+    /// that collect job does not exist, then a UUID is generated and the
+    /// collect job is stored, and the UUID is returned.
+    #[tracing::instrument(skip(self, encoded_aggregation_parameter), err)]
+    pub(crate) async fn lookup_collect_job_uri(
```

Comment:
IMO this shouldn't be named "lookup", since it creates collect jobs, too.

To match the API exposed by Transaction for other nounds, maybe create a pair of methods, "get_collect_job_for_parameters" (properly wordsmithed, returning NotFound if no collect job is returned), and "put_collect_job"? Ideally the getter would return a (new) CollectJob type that represents a collect job.